backend-refactor/OutOfDomainDetection.py

```diff
@@ -0,0 +1,6 @@
+class OutOfDomainDetection:
```

Comment:
Shouldn't the different pipeline steps be inherited from a common class?
```suggestion
class OutOfDomainDetection(PipelineStep):
```

Reply:
I created the common class which contains one abstract method